nixos/modules/services/web-servers/phpfpm/default.nix

```diff
@@ -287,6 +302,24 @@ in
           let
             cfgFile = fpmCfgFile pool poolOpts;
             iniFile = phpIni poolOpts;
+            wrapper = pkgs.writeShellApplication {
+              name = "phpfpm-exec-start-pre";
+              runtimeInputs = with pkgs; [
+                mount
+              ];
+              text = ''
+                if [ -v CREDENTIALS_DIRECTORY ]; then
+                  OLD_CREDENTIALS_DIRECTORY="$CREDENTIALS_DIRECTORY"
+                  export CREDENTIALS_DIRECTORY="/tmp''${CREDENTIALS_DIRECTORY}"
+                  mkdir -p "$CREDENTIALS_DIRECTORY"
+                  mount -t tmpfs -o size=1M,nodev,nosuid,noexec,mode=500,uid="$(id -u "${poolOpts.user}")",gid="$(id -g "${poolOpts.group}")" tmpfs "$CREDENTIALS_DIRECTORY"
```

Comment:
While systemd removes contents in here, have you also checked if the tmpfs is correctly unmounted?